Autofocus cancel button

draperunner opened the pull request:
Makes it easier using the popup with keyboard only. The cancel button is autofocused, so that pressing enter will close the popup. OK button is just one tap of the TAB key away. The reason the OK button is not autofocused is because accidentally pressing enter could then potentially cause harm.

draperunner committed:
Autofocus cancel button

Makes it easier using the popup with keyboard only. The cancel button is autofocused, so that pressing enter will close the popup. OK button is just one tap of the TAB key away. The reason the OK button is not autofocused is because accidentally pressing enter could then potentially cause harm.

draperunner committed:
Style buttons when focused similarly as when hovered

mathieudutour commented:
Thanks for the PR! Maybe we can add an option to choose which button to autofocus (with cancel by default). But there are a lot of use case when it's not harmful to confirm.

draperunner committed:
Add option for button autofocus

The focus property of the options object must be "ok" to focus okButton, "cancel" to autofocus the cancelButton, and any other non-empty string for none of the buttons to be focused. The default is "cancel". The _options object is added to this in order for _init to access the options. The README is updated.

draperunner commented:
Absolutely! That's a very good idea. Commit ee8f387e8db49d8ab6193e3bbb06aa5d7566b814 adds an option property to choose which button to focus. 

The focus property of the options object must be "ok" to focus okButton, "cancel" to autofocus the cancelButton (case-insensitive, "OK" for example, is fine). The default is "cancel". 

Any other non-empty string will result in none of the buttons being focused. That means one could write for example `focus: "none"` or `focus: "Things that actually matter"`.

I added  `_options` to `this` in order for the `._init()` method to access the options. The `.focus()` calls can not be done in the constructor, they must happen in `._init()`.

The README is updated.

mathieudutour commented:
Awesome!
